### PR TITLE
Yuri merge

### PR DESCRIPTION
**Pull request by garrettwrong**
unpacked most of the zip file and brought into git. Will continue review and fix conflicts before putting into master.

### CODE REVIEW

File: ampfar.f90 [hunk 1]
```diff
@@ -374,9 +374,12 @@ subroutine ampfsoln(it,nn) ! Called from tdchief
 
 
       do ll=1,lrz
-         ampfln(ll)=ampfarl(f_(0:ll,0,kelec,ll),ll)
-         ampflh(ll)=ampfarl(fh(0:ll,0,kelec,ll),ll)
-         ampflg(ll)=ampfarl(fg(0:ll,0,kelec,ll),ll)
+         ampfln(ll)=ampfarl(f_(0:iy+1,0:jx+1,kelec,ll),ll)
+         ampflh(ll)=ampfarl(fh(0:iy+1,0:jx+1,kelec,ll),ll)
+         ampflg(ll)=ampfarl(fg(0:iy+1,0:jx+1,kelec,ll),ll)
+         !YuP[2019-05-30] Note that fh() and fg() are allocated as
```
Comment (garrettwrong) [inline ampfar.f90]: excellent note, thank you.

File: impavnc0.f90 [hunk 2]
```diff
@@ -32,15 +32,15 @@ module impavnc0_mod
...
-  integer,pointer  :: k => advnce_k
+  !use advnce_mod  !here: in impavnc0_mod. To get k?
+  !integer,pointer  :: k => advnce_k
 
   ! gauss_, these are used in it3dalloc
   integer, public ::  inewjmax, ialloc
-  ! XXX this was an integer somewhere?
-  ! XXXvnc vs vnc0 abd etc?
+  ! XXX this was an integer somewhere?  ! YuP: ialloc is not used anymore.
+  ! XXX vnc vs vnc0 abd etc?   YuP: impavnc should be removed from compilation; only impavnc0 is used
```
Comment (garrettwrong) [inline impavnc0.f90]: okay i'll make a note to delete. thanks

File: impavnc0.f90 [hunk 3]
```diff
@@ -447,11 +453,20 @@ subroutine impavnc0(kopt)
        !-YuP      if (alloc.eq."enabled") then
        if(ASSOCIATED(ipivot)) then
           ! abd and ipivot are already allocated => do nothing
+          abd=zero !YuP[2019-04-24] was absent
```
Comment (garrettwrong) [inline impavnc0.f90]: yikes

File: mpilib.f90 [hunk 4]
```diff
@@ -8,34 +8,41 @@ module mpilib_mod
 
   !---END USE
 
+  !YuP[2019-05-31] These subroutines could be moved into mpins_par.f (in /mpi/)
```
Comment (garrettwrong) [inline mpilib.f90]: sounds good, less is more. note that i made a note to make your note into an issue.

File: tdtrvtor3.f90 [hunk 5]
```diff
@@ -65,7 +65,10 @@ subroutine tdtrvtor3(f1,f2,vp,vp_,kopt,k)
       leff=lrors
       if (kopt .eq. 3) leff=lrz
 
-      call dcopy (iyjx2*ngen*leff,f1,1,f2,1)
+      call dcopy(iyjx2*ngen*leff,f1(0:iy+1,0:jx+1,1:ngen,1:leff),1, &
+                                 f2(0:iy+1,0:jx+1,1:ngen,1:leff),1)
+      !YuP: Error? Why do we need to copy all 1:leff radial points
+      !     if the rest is done for one given ill=l_ point?
```
Comment (garrettwrong) [inline tdtrvtor3.f90]: you tell me..

File: wpadvnc.h [hunk 6]
```diff
@@ -15,15 +15,15 @@
...
-       zero,ilpm1ef(i,l,+1).ne.-999 .and. ilpm1ef(i,l,-1).ne.-999)
+!      wpa(i,j,k,l)=wpweigt(i,j,l)*(fsmid(i,j,k,l) - &
+!       fsmid(ilpm1ef(i,l,-1),j,k,lpm1eff(l,-1)))
+!      wpap(i,j,k,l)=cvmgt(wpweigp(i,j,l) * &
+!       (fnp1(ilpm1ef(i,l,+1),j,k,lpm1eff(l,+1))-fnp1(i,j,k,l)), &
+!       zero,ilpm1ef(i,l,+1).ne.-999)
+!      wpacen(i,j,k,l)=cvmgt(wpweigt(i,j,l) * &
+!       (fnp1(ilpm1ef(i,l,+1),j,k,lpm1eff(l,+1)) - &
+!       fnp1(ilpm1ef(i,l,-1),j,k,lpm1eff(l,-1))) , &
+!       zero,ilpm1ef(i,l,+1).ne.-999 .and. ilpm1ef(i,l,-1).ne.-999)
```
Comment (garrettwrong) [inline wpadvnc.h]: what is this about

File: impavnc0.f90 (hunk 2 above)
Comment (garrettwrong) [reply impavnc0.f90]: deleted

File: mpilib.f90 (hunk 4 above)
Comment (garrettwrong) [reply mpilib.f90]: see #18

**Comment by ypetrov** [comment]
YuP: Those statement functions that are commented out are not used 
anywhere.

File: wpadvnc.h (hunk 6 above)
Comment (garrettwrong) [reply wpadvnc.h]: deleted per yuri

**Pull request merged**